backend/danswer/chat/process_message.py

```diff
@@ -271,6 +275,66 @@ def stream_chat_message(
                 doc_identifiers=identifier_tuples,
                 document_index=document_index,
             )
+
+            # truncate the last document if it exceeds the token limit
+            max_document_tokens = compute_max_document_tokens(
+                persona, actual_user_input=message_text
+            )
+            tokens_per_doc = [
+                len(
+                    llm_tokenizer_encode_func(
+                        build_doc_context_str(
```

Comment:
Is this recounting the content? Do we already know how many tokens that is, the tokenizer is quite slow so we should avoid any recounting possible. But... maybe doesn't really matter

Reply:
I'd rather just make a comment of it if it is the case and just move on